src/regex/pcre2.cr

```diff
@@ -68,6 +68,7 @@ module Regex::PCRE2
         flag |= case option
                 when .ignore_case?       then LibPCRE2::CASELESS
                 when .multiline?         then LibPCRE2::DOTALL | LibPCRE2::MULTILINE
+                when .multiline_only?    then LibPCRE2::MULTILINE
```

Comment:
**suggestion:** If `MULTILINE_ONLY | DOTALL == MULTILINE` we can drop handling `MULTILINE` here because both `MULTILINE_ONLY` and `DOTALL` are mapped. Ditto for `pcre_compile_options`.

Reply:
Seems to work!